### PR TITLE
OEUI-65: Optimising OEUI-56

### PR DESCRIPTION
**Pull request by fuchodeveloper**
## JIRA TICKET NAME

[OEUI-65: Optimising OEUI-56](https://issues.openmrs.org/browse/OEUI-65)

## SUMMARY
This involves optimizing the code of a previous ticket [OEUI-56](https://issues.openmrs.org/projects/OEUI/issues/OEUI-56)

### CODE REVIEW

File: app/js/actions/careSetting.js [hunk 1]
```diff
@@ -1,51 +1,29 @@
...
 
 const contextPath = window.location.href.split('/')[3];
 const apiBaseUrl = `/${contextPath}/ws/rest/v1`;
 
-export function fetchInpatientCareSetting() {
-  return dispatch => axios.get(`${apiBaseUrl}/caresetting?q=inpatient`)
+const fetchPatientCareSetting = () => dispatch =>
+  axios.get(`${apiBaseUrl}/caresetting`)
     .then((response) => {
       dispatch({
```
Comment (larrystone) [inline app/js/actions/careSetting.js]: Looks like there is no loading action logic implemented here, is it intention?
I guess the team also agreed on not allowing the dispatch action direction return its objects

File: app/js/components/orderEntry/OrderEntryPage.jsx [hunk 2]
```diff
@@ -2,44 +2,49 @@ import React from 'react';
...
 
-class OrderEntryPage extends React.Component {
+export class OrderEntryPage extends React.Component {
   componentDidMount() {
-    this.props.fetchInpatientCareSetting();
-    this.props.fetchOutpatientCareSetting();
+    this.props.fetchPatientCareSetting();
   }
   render() {
+    if (!(this.props.outpatientCareSetting || this.props.inpatientCareSetting)) {
```
Comment (larrystone) [inline app/js/components/orderEntry/OrderEntryPage.jsx]: Correct me if i'm wrong, but his algorithm implies that if outpatientCareSetting and/or inpatientCareSetting has a valid value (1), it should show the loader...
```
!(1 + 0) = !1 = 0
!(1 + 1) = !1 = 0
!(0 + 0) = !0 = 1
```
I'm thinking it should be 
```
(!this.props.outpatientCareSetting || !this.props.inpatientCareSetting)
or 
!(this.props.outpatientCareSetting && this.props.inpatientCareSetting) - distributive law of logic
```

Comment (fuchodeveloper) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: Will look into this.

Comment (fuchodeveloper) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: Had an initial implementation using `(!this.props.outpatientCareSetting || !this.props.inpatientCareSetting)`, while refactoring the code further to include both cases with a single negation, did not replace the `||`.
Thanks for pointing this out.

Comment (fuchodeveloper) [reply app/js/components/orderEntry/OrderEntryPage.jsx]: Implementation updated now.

File: app/js/actions/careSetting.js (hunk 1 above)
Comment (fuchodeveloper) [reply app/js/actions/careSetting.js]: Code updated to reflect loading and action creator dispatch.

File: app/js/actions/careSetting.js [hunk 3]
```diff
@@ -1,51 +1,43 @@
 import axios from 'axios';
 import {
-  INPATIENT_CARESETTING,
-  INPATIENT_CARESETTING_ERROR,
-  OUTPATIENT_CARESETTING,
-  OUTPATIENT_CARESETTING_ERROR,
+  PATIENT_CARESETTING,
```
Comment (Betty-Kebenei) [inline app/js/actions/careSetting.js]: Could you please make this more descriptive? Say `PATIENT_CARESETTING_SUCCESS`

File: app/js/components/orderEntry/OrderEntryPage.jsx [hunk 4]
```diff
@@ -2,44 +2,49 @@ import React from 'react';
...
-  fetchOutpatientCareSetting: PropTypes.func.isRequired,
+  fetchPatientCareSetting: PropTypes.func.isRequired,
+  location: PropTypes.shape({}).isRequired,
+  outpatientCareSetting: PropTypes.shape({}).isRequired,
 };
 
 const mapStateToProps = ({ careSettingReducer }) => ({
-  inpatientCareSetting: careSettingReducer.inpatientCareSetting,
   outpatientCareSetting: careSettingReducer.outpatientCareSetting,
+  inpatientCareSetting: careSettingReducer.inpatientCareSetting,
```
Comment (Betty-Kebenei) [inline app/js/components/orderEntry/OrderEntryPage.jsx]: Could you destructure this further?

File: app/js/actions/careSetting.js (hunk 3 above)
Comment (fuchodeveloper) [reply app/js/actions/careSetting.js]: That makes sense.
Would notify the team about this suggestions and make changes based on the team's decision.

Comment (fuchodeveloper) [reply app/js/actions/careSetting.js]: Have implemented new changes based on feedback from the team. The action type is now `PATIENT_CARESETTING_SUCCESS`.

**Pull request merged**